### DISCUSSION TITLE
Issue with installing sleap=1.2.9 from github

### DISCUSSION THREAD

**Discussion by met4t** [talmolab/sleap#1087] [Help!] [answered]
Hello,

I have attempted to install sleap v 1.2.9 from github following the installation instructions provided. 
I run into the error that is printed below.

ERROR conda.core.link:_execute(730): An error occurred while installing package 'conda-forge::qt-5.12.9-h556501e_6'.

 I, thus, cannot install SLEAP. What is best for met to do ?

**Comment by roomrys** [comment, marked as answer]
Hi @met4t,

It could be a problem with an outdated conda (similar external [post](https://github.com/conda/conda/issues/8661)). Can you try updating your conda with:
```
conda update -y -n base -c defaults conda
```
Then, before attempting to (re)create the conda environemnt, make sure that the environment you are attempting to create (specified by the name that follows the `-n` argument, i.e. `conda env create -n <name_of_environement>`) does not exist. You can view all environments with
```
conda env list
```
and then if the environment you are attempting to create exists, you can remove it with
```
conda env remove -n <name_of_environment>
```
e.g. `conda env remove -n sleap`.

Finally, open a terminal in the "sleap" directory that contains all the source code (wherever you git cloned the source code to). Now you can create the conda environment ([steps 6&7 for m1](https://sleap.ai/installation.html#m1-macs) or [step 3 for other](https://sleap.ai/installation.html#conda-from-source)).

If you are still having trouble, do you mind providing:
- the full command + error message
- your OS (I am assuming it's an m1)

Thanks,
Liezl

**Comment by met4t** [comment]
Thanks, Liezl for this followup.

When removing the environments, should I remove the Miniconda3 environment
as well ?
I wish to recall that I first installed the Miniconda3 package in my
Windows.
I am running the git commands from the resulting Anaconda Prompt.

Cheers,
Marshall

**Comment by met4t** [comment]
I have followed your instructions on updating conda. But still an error is
generated, which I have fully provided.

This is the command I use, after removing the environments and updating
conda:

```
(base) 1.2.9) ==1.2.9) C:\Miniconda3\sleap>conda env create -f
environment_no_cuda.yml -n sleap=1.2.9

Warning: you have pip-installed dependencies in your environment file, but
you do not list pip itself as one of your conda dependencies.  Conda may
not use the correct pip to install your packages, and they may end up in
the wrong place.  Please add an explicit pip dependency.  I'm adding one
for you, but still nagging you.
Collecting package metadata (repodata.json): done
Solving environment: done
```

You can notice there is a warning message after the command line, telling
me that pip dependencies are not listed.

This is the end of the message I get from running that:

```
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
ERROR conda.core.link:_execute(740): An error occurred while installing
package 'conda-forge::qt-5.12.9-h556501e_6'.
Rolling back transaction: done

LinkError: post-link script failed for package
conda-forge::qt-5.12.9-h556501e_6
location of failed script:
C:\Miniconda3\envs\sleap=1.2.9\Scripts\.qt-post-link.bat
==> script messages <==
<None>
==> script output <==
stdout:
stderr: 'C:\Miniconda3\envs\sleap' is not recognized as an internal or
external command,
operable program or batch file.

return code: 1

()
```

**Comment by roomrys** [comment]
Hi @met4t,

So I tested this on my machine, and verified that conda dislikes the environment name `-n sleap=1.2.9`. Can you try a name that just contains letters, i.e. `conda env create -f environment_no_cuda.yml -n sleap`? I think it is the equals sign in the name which is throwing things off.

Thanks,
Liezl

**Comment by met4t** [comment]
Hi Liezl,

I had tried 'sleap' instead of 'sleap=1.2.9' several days ago, but what it
did is try to install all available versions of sleap.
Then, it ran into conflict in the dependencies required for the different
versions.

I will try it again, and post you the message.

**Comment by met4t** [comment]
I have used the command below to install the conda environment, and it went
well:

```
(base) C:\Miniconda3\sleap>conda env create -f environment_no_cuda.yml -n sleap
```

The next step is to do a pip install of sleap. If I use the command below,
you think everything will go well ?

```
pip install sleap
```

**Reply by roomrys** [reply]
You shouldn't have to `pip install sleap` if you successfully created the conda environment. Just `conda activate sleap` and run `sleap-label` to open the GUI. (The `sleap` package is already available from within the conda environment)

**Comment by met4t** [comment]
The GUI opened. Many Thanks !
